src/util/cookieString.ts

```diff
@@ -42,10 +42,13 @@ export function encodeCookieString(obj: Record<string, string>): string {
 export function merge(old: string, obj: Record<string, string>): Record<string, string> {
     const prevData = parseCookieString(old);
 
-    // Delete `point` keys to avoid inconsistencies like having a `pn_alias` together with a `pn_root`,
-    delete prevData.pn_root;
-    delete prevData.pn_routes;
-    delete prevData.pn_alias;
+    // If there's a pn_alias, there cannot be pn_root + pn_routes (and viceversa)
+    if (obj.pn_alias) {
```

Comment:
Sorry I missed from the beginning. The function name is `merge` which makes it unclear it may omit some keys in the resulting object. Might be confusing for other devs. Let's add an intermediary function or rename this one to `mergeAndResolveRecordConflicts` or something like that.

Reply:
> Might be confusing for other devs.

Hopefully, the time I take to annotate functions with comments helps other developers.

Reply:
@germanviescas I really appreciate the time you take to annotate the code you provide, it really helps in most of the cases.

But please consider my point of view. If I see a function name like `merge` which is a super-basic operation, it would never occur to me that I have to go and check its' source code. Just because I expect it to be something very simple.

Now think of it a bit further. Imagine one has to make a super-urgent hot fix that involves using this function. In this case most of the devs would probably think it must be a library function.

Most likely not only they won't check the source code, they even wouldn't bother testing its' result, which will of course cause the hot fix release to be buggy.

Therefore please don't get irritated by comments like that, I am just trying to cover some extreme and rare cases that still can sometimes happen.